keyboards/anavi/macropad12/macropad12.c

```diff
@@ -0,0 +1,37 @@
+// Copyright 2023 Leon Anavi <leon@anavi.org>
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "quantum.h"
+#include <stdio.h>
+
+#ifdef OLED_ENABLE
+
+bool oled_task_kb(void) {
+
+    if (!oled_task_user()) {
+        return false;
+    }
+
+    // Host Keyboard Layer Status
+    oled_write_ln_P(PSTR("ANAVI Macro Pad 12"), false);
+    oled_write_ln_P(PSTR("Keymap: Default"), false);
+
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(PSTR("Num Lock: "), false);
+    oled_write_ln_P(led_state.num_lock ? PSTR("On") : PSTR("Off"), false);
+    oled_write_P(PSTR("Caps Lock: "), false);
+    oled_write_ln_P(led_state.caps_lock ? PSTR("On") : PSTR("Off"), false);
+    oled_write_P(PSTR("Scroll Lock: "), false);
+    oled_write_ln_P(led_state.scroll_lock ? PSTR("On") : PSTR("Off"), false);
+#   ifdef RGBLIGHT_ENABLE
+        static char rgbStatusLine1[26] = {0};
+        snprintf(rgbStatusLine1, sizeof(rgbStatusLine1), "RGB Mode: %d", rgblight_get_mode());
+        oled_write_ln(rgbStatusLine1, false);
+        static char rgbStatusLine2[26] = {0};
+        snprintf(rgbStatusLine2, sizeof(rgbStatusLine2), "h:%d s:%d v:%d", rgblight_get_hue(), rgblight_get_sat(), rgblight_get_val());
+        oled_write_ln(rgbStatusLine2, false);
```

Comment:
It's a bit more verbose, but this can be accomplished without snprintf (which I generally recommend against, since copy-paste happens)
```suggestion
        oled_write_P(PSTR("RGB Mode: "), false);
        oled_write_ln(get_u8_str(rgblight_get_mode(), ' '), false);
        oled_write_P(PSTR("h: "), false);
        oled_write(get_u8_str(rgblight_get_hue(), ' '), false);
        oled_write_P(PSTR("s: "), false);
        oled_write(get_u8_str(rgblight_get_sat(), ' '), false);
        oled_write_P(PSTR("v: "), false);
        oled_write_ln(get_u8_str(rgblight_get_val(), ' '), false);
```

Alternatively, if you wanted to print the actual name for the rgb modes: 
https://github.com/qmk/qmk_firmware/blob/master/users/drashna/rgb/rgb_stuff.c#L137-L165
https://github.com/qmk/qmk_firmware/blob/master/users/drashna/oled/oled_stuff.c#L484-L504

Taken lovingly from tzarc's code.

Reply:
Thanks, I modified the code as per your recommendations. I prefer using a number instead of the actual name for the rgb mode because the screen is small.